pkgs/sops-install-secrets/main.go

```diff
@@ -330,6 +335,21 @@ func mountSecretFs(mountpoint string, keysGid int) error {
 	return nil
 }
 
+func remountSecretFsRo(mountpoint string) error {
```

Comment:
To reduce de-duplicate logic, I would propose this

```suggestion
func ensureSecretFs(mountpoint string, readonly bool) error {
}
```

This function would than use `Statfs` and to check if we need to mount it, or if we need to remount as read-only/read-write.